samples/bluetooth/iso_connected_benchmark/src/main.c

```diff
@@ -1044,8 +1049,83 @@ static int change_central_settings(void)
 	return 0;
 }
 
+static uint16_t calc_conn_interval(const struct bt_iso_cig_param *cig_param)
+{
+	/* Convert from microseconds to connection interval units (1.25ms)*/
+	const uint32_t c_to_p_interval = BT_GAP_US_TO_CONN_INTERVAL(cig_param->c_to_p_interval);
+	const uint32_t p_to_c_interval = BT_GAP_US_TO_CONN_INTERVAL(cig_param->p_to_c_interval);
+	const uint32_t max_conn_interval = BT_GAP_MS_TO_CONN_INTERVAL(500U);
+	const uint32_t min_conn_interval = BT_GAP_INIT_CONN_INT_MIN;
+	uint32_t tmp_c_to_p_interval = c_to_p_interval;
+	uint32_t tmp_p_to_c_interval = p_to_c_interval;
+	uint32_t tmp_common_interval = 0U;
+	uint32_t conn_interval;
+
+#if defined(CONFIG_BT_ISO_TEST_PARAMS)
+	tmp_common_interval = MIN(cig_param->iso_interval, max_conn_interval);
+#endif /* CONFIG_BT_ISO_TEST_PARAMS */
+
+	if (tmp_common_interval == 0U) {
+		/* Attempt to find a common SDU interval lower than max_conn_interval to use as base
+		 * for connection interval
+		 */
+		while (tmp_c_to_p_interval != tmp_p_to_c_interval &&
+		       MAX(tmp_c_to_p_interval, tmp_p_to_c_interval) < max_conn_interval) {
+			if (tmp_c_to_p_interval < tmp_p_to_c_interval) {
+				tmp_c_to_p_interval += c_to_p_interval;
+			} else {
+				tmp_p_to_c_interval += p_to_c_interval;
+			}
+		}
+
+		/* Will either be a common value or max_conn_interval */
+		tmp_common_interval =
+			MIN(MAX(tmp_c_to_p_interval, tmp_p_to_c_interval), max_conn_interval);
+	}
+
+	if (tmp_common_interval == max_conn_interval) {
+		conn_interval = tmp_common_interval;
+	} else {
+		/* Default connection interval to 6 times the SDU interval and reduce to a multiple
+		 * less than max_conn_interval_us
+		 */
+		conn_interval = tmp_common_interval * 6U;
+
+		/* Ensure that the conn interval is between min_conn_interval and
+		 * max_conn_interval for the sake of interopability and scheduling
+		 */
+		while (conn_interval < min_conn_interval) {
+			conn_interval += tmp_common_interval;
+		}
+		while (conn_interval > max_conn_interval) {
+			conn_interval -= tmp_common_interval;
+		}
+	}
+
+	/* If we cannot convert back then it's not a lossless conversion */
+	if (BT_CONN_INTERVAL_TO_US(c_to_p_interval) != cig_param->c_to_p_interval) {
+		LOG_WRN("c_to_p_interval of %u us is not a multiple of the connection interval "
+			"unit (1.25 ms) and may be subpar. Suggest to adjust SDU "
+			"interval to be a multiple of 1.25 ms",
+			cig_param->c_to_p_interval);
+	}
+
+	/* If we cannot convert back then it's not a lossless conversion */
+	if (BT_CONN_INTERVAL_TO_US(p_to_c_interval) != cig_param->p_to_c_interval) {
+		LOG_WRN("p_to_c_interval of %u us is not a multiple of the connection interval "
+			"unit (1.25 ms) and may be subpar. Suggest to adjust SDU "
+			"interval to be a multiple of 1.25 ms",
+			cig_param->p_to_c_interval);
+	}
+
+	return conn_interval;
+}
+
 static int central_create_connection(void)
 {
+	const uint16_t conn_interval = calc_conn_interval(&cig_create_param);
```

Comment:
Some controllers are able to select a "good" connection interval if the connection interval range passed to the controller is sufficiently large. Can the zephyr controller do that?

That would result in a simpler implementation

Reply:
At this point the controller would be unaware of any future ISO connections, and can thus not decide on a good interval that works with the future and unknown ISO interval, right?

Reply:
>  the controller can select an ISO interval which fits with the other activities.

To some extend; for unframed ISO the ISO interval shall be a multiple of the SDU inteval. 

But are you suggesting that I revert this change, and leave it up to the controller? 
Alternatively we can also ask the user for the conn parameters, as we do for the ISO parameters

Reply:
Could a generic solution instead be:

1) Create CIG
2) Connect ACL
3) Connect CIS

This sample is today doing 
1) Connect ACL
2) Create CIG
3) Connect CIS

Would it help creating the CIG before establishing the ACL to help the controller pick reasonable ACL intervals? 
Or is it better to create the ACL first so that the controller can pick reasonable ISO intervals, assuming that the test params are not used? 

Ideally the host should just sending a reasonable interval range and have the controller do the intervals, as that is (or should be) controller agnostic